core/src/main/java/org/elasticsearch/action/admin/indices/warmer/get/GetWarmersResponse.java

```diff
@@ -94,7 +94,7 @@ public void writeTo(StreamOutput out) throws IOException {
             for (IndexWarmersMetaData.Entry warmerEntry : indexEntry.value) {
                 out.writeString(warmerEntry.name());
                 out.writeStringArray(warmerEntry.types());
-                out.writeBytesReference(warmerEntry.source());
+                warmerEntry.source().writeTo(out);
```

Comment:
I see that source() is marked `@Nullable`, should we then check that it's not null here?